### PR TITLE
[backend/frontend] Added support for Consent Message on Login Page

### PR DESCRIPTION
**Pull request by ParamConstructor**
### Proposed changes

This PR allows for a customizable (off settings GUI by Admin) "Consent to Monitor" or "Legal Notice" message ('platform_consent_message') that is displayed beneath the 'platform_login_message' (if configured) on the Login Page. If the new ('platform_consent_message') is configured, the message will be displayed on the login page, and the user will be required to click a consent check box agreeing to the displayed terms, before the Login Form is displayed.

If the new ('platform_consent_message') is blank in the configuration, the login page will ignore this field and display nothing.

### Related issues

This PR has been tested with LDAP configuration, standard internal OpenCTI authentication, and SSO SAML authentication. 

### Checklist

- [ X ] I consider the submitted work as finished
- [ X ] I tested the code for its functionality
- [ X ] I wrote test cases for the relevant uses case
-          Tested manually within the GUI. A future PR will have Selenium autotest to supplement this capability automatically.
- [ ] I added/update the relevant documentation (either on github or on notion)
-          This has not been done, suspect the ('platform_consent_message')  option needs to be added to Notion page documentation.
- [ X ] Where necessary I refactored code to improve the overall quality

### Further comments

None.

### CODE REVIEW

File: opencti-platform/opencti-front/src/private/components/settings/Settings.js [hunk 1]
```diff
@@ -416,6 +420,28 @@ const Settings = () => {
                               variant="standard"
                               helperText={<SubscriptionFocus context={editContext} fieldName="platform_login_message"/>}
                             />
+                            <Typography variant="h1" gutterBottom={true} style={{ marginTop: '20px' }}>
+                              {t('Platform Consent Message')}
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/components/settings/Settings.js]: You are missing some Localization translation in the files.

Additionnaly, in this screen we do not have such "h1" title, maybe you can remove it ?

File: opencti-platform/opencti-front/src/private/components/settings/Settings.js [hunk 2]
```diff
@@ -416,6 +420,28 @@ const Settings = () => {
...
+                            </Typography>
+                            <Field
+                              component={MarkDownField}
+                              name="platform_consent_message"
+                              label={t('Platform Consent Message - requires acceptance to enable login form when set')}
+                              fullWidth={true}
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/components/settings/Settings.js]: You don't need to add ={true} in React component, it's redundant. I know we have done it **a lot** in the code, but you can remove it from this part :)

File: opencti-platform/opencti-front/src/public/components/Login.tsx [hunk 3]
```diff
@@ -177,7 +184,11 @@ const Login: FunctionComponent<LoginProps> = ({ type, settings }) => {
     loginHeight = 150;
   }
   const marginTop = dimension.height / 2 - loginHeight / 2 - 200;
-
+  const [checked, setChecked] = useState(false);
+  const handleChange = () => {
+    setChecked(!checked);
+  };
+  const isConsentMessage = consentMessage && consentMessage.length > 0;
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/public/components/Login.tsx]: I think you can use the isEmptyField utils we already have for this.

File: opencti-platform/opencti-front/src/public/components/Login.tsx [hunk 4]
```diff
@@ -190,8 +201,20 @@ const Login: FunctionComponent<LoginProps> = ({ type, settings }) => {
           <Markdown>{loginMessage}</Markdown>
         </Paper>
       )}
-      {isAuthForm && (
-        <Paper variant="outlined">
+      {consentMessage && consentMessage.length > 0 && (
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/public/components/Login.tsx]: You should use the constant previously created

File: opencti-platform/opencti-front/src/public/components/Login.tsx [hunk 5]
```diff
@@ -190,8 +201,20 @@ const Login: FunctionComponent<LoginProps> = ({ type, settings }) => {
...
+        <Paper classes={{ root: classes.paper }} variant="outlined">
+          <Markdown>{consentMessage}</Markdown>
+          <Markdown> I have read and comply with the above statement: </Markdown>
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/public/components/Login.tsx]: Why you add this message hardcoded way and you do not set it as a default value in the message ?

File: opencti-platform/opencti-front/src/public/components/Login.tsx [hunk 6]
```diff
@@ -12,12 +12,18 @@ import OTPForm from './OTPForm';
 import OtpActivationComponent from './OtpActivation';
 import { Theme } from '../../components/Theme';
 import { LoginRootPublicQuery$data } from '../__generated__/LoginRootPublicQuery.graphql';
+import Checkbox from '@mui/material/Checkbox';
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/public/components/Login.tsx]: Linter error here

File: opencti-platform/opencti-front/src/public/components/Login.tsx [hunk 7]
```diff
@@ -190,8 +201,20 @@ const Login: FunctionComponent<LoginProps> = ({ type, settings }) => {
           <Markdown>{loginMessage}</Markdown>
         </Paper>
       )}
-      {isAuthForm && (
-        <Paper variant="outlined">
+      {consentMessage && consentMessage.length > 0 && (
+        <Paper classes={{ root: classes.paper }} variant="outlined">
+          <Markdown>{consentMessage}</Markdown>
+          <Markdown> I have read and comply with the above statement: </Markdown>
+          <Checkbox name='consent' edge='start' onChange={handleChange} style={{ margin: 0 }}></Checkbox>
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/public/components/Login.tsx]: Could you try to align this checkbox with the line of text of compliance ?

**Review by Kedae — CHANGES_REQUESTED** [review]

File: opencti-platform/opencti-front/src/private/components/settings/Settings.js (hunk 1 above)
Comment (ParamConstructor) [reply opencti-platform/opencti-front/src/private/components/settings/Settings.js]: @Kedae Translation Request here is to add to the various translations for the text of "Platform Consent Message" and others that are wrapped with t() into opencti-platform/opencti-front/src/utils/Localization.js?

File: opencti-platform/opencti-front/src/public/components/Login.tsx (hunk 4 above)
Comment (ParamConstructor) [reply opencti-platform/opencti-front/src/public/components/Login.tsx]: @Kedae - replace with isConsentMessage constant - sorry about that....

File: opencti-platform/opencti-front/src/private/components/settings/Settings.js (hunk 1 above)
Comment (Kedae) [reply opencti-platform/opencti-front/src/private/components/settings/Settings.js]: Yes exactly. We use i18n to translate the app and the Localization.js file contains all the translation. So you need to add all the new keys within t() inside this file for every language. Basically we use Google to translate the text (and obviously we will be able to help you on the French one :) )

File: opencti-platform/opencti-front/src/public/components/Login.tsx (hunk 5 above)
Comment (ParamConstructor) [reply opencti-platform/opencti-front/src/public/components/Login.tsx]: @Kedae - will correct and make part of opencti-platform/opencti-front/src/utils/Localization.js as a string.

File: opencti-platform/opencti-front/src/public/components/Login.tsx (hunk 7 above)
Comment (ParamConstructor) [reply opencti-platform/opencti-front/src/public/components/Login.tsx]: @Kedae Placed in Box tags and aligned as requested.

File: opencti-platform/opencti-front/src/public/components/Login.tsx (hunk 6 above)
Comment (ParamConstructor) [reply opencti-platform/opencti-front/src/public/components/Login.tsx]: @Kedae check-ts and lint have been run to fix any errant issues. Should be resolved on current commit.

File: opencti-platform/opencti-front/src/public/components/Login.tsx (hunk 5 above)
Comment (ParamConstructor) [reply opencti-platform/opencti-front/src/public/components/Login.tsx]: @Kedae A new graphql field has been added (platform_consent_confirm_text). This is by default, if left blank on the Setting page, set to "I have read and comply with the above statement". Translations have been added for the default string. An Admin can set a new message in the Setting page to replace this default message/string.

Is this agreeable?

File: opencti-platform/opencti-front/src/public/components/Login.tsx (hunk 4 above)
Comment (ParamConstructor) [reply opencti-platform/opencti-front/src/public/components/Login.tsx]: @Kedae - Fixed to use constant (isConsentMessage).

File: opencti-platform/opencti-front/src/private/components/settings/Settings.js (hunk 2 above)
Comment (ParamConstructor) [reply opencti-platform/opencti-front/src/private/components/settings/Settings.js]: @Kedae - Fixed to just read 'fullWidth' versus 'fullWidth={true}' for calls within file.

File: opencti-platform/opencti-front/src/private/components/settings/Settings.js (hunk 1 above)
Comment (ParamConstructor) [reply opencti-platform/opencti-front/src/private/components/settings/Settings.js]: @Kedae - I believe this is all of them that are required. Added translations for:      
- 'Platform Consent Message'
- 'Platform Consent Message - requires acceptance to enable login form when set'
- 'I have read and comply with the above statement'
- 'Platform Consent Confirm Text - confirm label next to confirm checkbox'

h1 tag replaced with proper h4 tag based on other elements within this file.

Comment (ParamConstructor) [reply opencti-platform/opencti-front/src/private/components/settings/Settings.js:428]: @Kedae - I believe this is all of them that are required. Added translations for:      
- 'Platform Consent Message'
- 'Platform Consent Message - requires acceptance to enable login form when set'
- 'I have read and comply with the above statement'
- 'Platform Consent Confirm Text - confirm label next to confirm checkbox'

File: opencti-platform/opencti-front/src/public/components/Login.tsx (hunk 3 above)
Comment (ParamConstructor) [reply opencti-platform/opencti-front/src/public/components/Login.tsx]: @Kedae - Fixed with isNotEmptyField() from utils.

**Comment by richard-julien** [comment]
Hi @ParamConstructor,

Some remarks on the PR.

- The confirm button doesn't hide all the SSO buttons for now, i think sso should also be hidden until validation
- UI configuration is now a bit unbalance between left side and right side, we need to rework a bit the screen. There is also some problem in change detection some times, that is not related to your work but maybe could be interesting to also take care here.
- Do you think it could be possible for you to activate https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork ? Could be very interesting to be able for us to also directly commit on your branch for this kind of remarks that will be quicker to just fix :)

**Comment by ParamConstructor** [comment]
@richard-julien - The "Allow edits by maintainers" feature apparently only works if the forked repo is not part of an organization :-(. They have an open bug on this - https://github.com/orgs/community/discussions/5634 I tried to enable it - but the checkbox is not even displayed on my PRs to enable (as referenced in the open issue). In this PR, we had not wrapped the SSO method to have to consent - but I can make that update.

The "unbalanced" setting screen will correct itself when you accept/merge the PR for the header/footer stuff. (#3302 ). The Header/Footer configuration options are in the left side where the unbalance exists - making the setting screen back to "even" distribution of configuration options.

**Comment by ParamConstructor** [comment]
@richard-julien @Kedae - SSO SAML has been tested. Support for SSO and removing from the login page before consent has been accepted has been added to the PR. PR has been rebased to master, as of the time of this comment.

**Pull request merged**